osu.Game/Screens/Play/EpilepsyWarning.cs

```diff
@@ -0,0 +1,107 @@
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.Play
+{
+    public class EpilepsyWarning : VisibilityContainer
+    {
+        private readonly BindableDouble trackVolumeOnEpilepsyWarning = new BindableDouble(1f);
+
+        private Track track;
+        private FillFlowContainer warningContent;
+
+        public EpilepsyWarning()
+        {
+            RelativeSizeAxes = Axes.Both;
+            Alpha = 0f;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours, IBindable<WorkingBeatmap> beatmap)
+        {
+            Children = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.Black.Opacity(0.5f),
+                },
+                warningContent = new FillFlowContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Direction = FillDirection.Vertical,
+                    Children = new Drawable[]
+                    {
+                        new SpriteIcon
+                        {
+                            Colour = colours.Yellow,
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Icon = FontAwesome.Solid.ExclamationTriangle,
+                            Size = new Vector2(50),
+                        },
+                        new OsuTextFlowContainer(s => s.Font = OsuFont.GetFont(size: 25))
+                        {
+                            RelativeSizeAxes = Axes.X,
+                            AutoSizeAxes = Axes.Y,
+                            TextAnchor = Anchor.Centre,
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                        }.With(tfc =>
+                        {
+                            tfc.AddText("This beatmap contains scenes with ");
+                            tfc.AddText("rapidly flashing colours", s =>
+                            {
+                                s.Font = s.Font.With(weight: FontWeight.Bold);
+                                s.Colour = colours.Yellow;
+                            });
+                            tfc.AddText(".");
+
+                            tfc.NewParagraph();
+                            tfc.AddText("Please take caution if you are affected by epilepsy.");
+                        }),
+                    }
+                }
+            };
+
+            track = beatmap.Value.Track;
+            track.AddAdjustment(AdjustableProperty.Volume, trackVolumeOnEpilepsyWarning);
+        }
+
+        protected override void PopIn()
+        {
+            this.FadeIn(500, Easing.InQuint)
+                .TransformBindableTo(trackVolumeOnEpilepsyWarning, 0.25, 500, Easing.InQuint);
```

Comment:
what's the reasoning for this? just to "look/sound cool"? just want to make sure i'm not missing something

Reply:
more like for getting serious attention from the user, but yeah nothing else than that.

Reply:
I liked it as it clearly diverted user attention, but I can't say I'm too attached to it myself. Same thing with the dim, really.